GEN_ABSOLUTE_SYM cannot handle value larger than INT_MAX on qemu_x86_64

If a kconfig (e.g. `CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC` is defined with a value > +2147483647, it fails to compile with following error:

```
In file included from ../include/toolchain.h:39,
                 from zephyr/misc/generated/configs.c:7:
zephyr/misc/generated/configs.c: In function '_ConfigAbsSyms':
../include/toolchain/gcc.h:404:2: error: invalid 'asm': operand is not a condition code, invalid operand code 'c'
  404 |  __asm__(".globl\t" #name "\n\t.equ\t" #name \
      |  ^~~~~~~
zephyr/misc/generated/configs.c:14:1: note: in expansion of macro 'GEN_ABSOLUTE_SYM'
   14 | GEN_ABSOLUTE_SYM(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC, 2147483648);
      | ^~~~~~~~~~~~~~~~
```

I haven't tried negative maximum but it is reasonable to believe that it would also fail to compile.

> Note that this fails only on `qemu_x86_64`, but works on `qemu_x86`.

It passes if the value is less than UINT_MAX, if it's bigger than that we get an overflow warning instead, it still builds but the value is truncated:

```
/home/apboie/projects/zephyr/zephyr/tests/kernel/mem_protect/userspace/out/zephyr/include/generated/autoconf.h:3:44: warning: overflow in conversion from 'long long int' to 'int' changes value from '9000000000' to '410065408' [-Woverflow]
    3 | #define CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC 9000000000
      |                                            ^~~~~~~~~~
/home/apboie/projects/zephyr/zephyr/kernel/timeout.c:33:33: note: in expansion of macro 'CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC'
   33 | int z_clock_hw_cycles_per_sec = CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC;
      |                      
```

--- a/include/toolchain/gcc.h
+++ b/include/toolchain/gcc.h
@@ -398,7 +398,14 @@ do {                                                                    \
 		",%B0"                              \
 		"\n\t.type\t" #name ",%%object" :  : "n"(~(value)))
 
-#elif defined(CONFIG_X86) || defined(CONFIG_ARC) || defined(CONFIG_ARM64)
+#elif defined(CONFIG_X86)
+
+#define GEN_ABSOLUTE_SYM(name, value)               \
+	__asm__(".globl\t" #name "\n\t.equ\t" #name \
+		",%p0"                              \
+		"\n\t.type\t" #name ",@object" :  : "n"(value))
+
+#elif defined(CONFIG_ARC) || defined(CONFIG_ARM64)
 
 #define GEN_ABSOLUTE_SYM(name, value)               \
 	__asm__(".globl\t" #name "\n\t.equ\t" #name \

